inc/interactive/class-phet.php

```diff
@@ -47,17 +77,32 @@ public function registerEmbedHandlerForExport() {
 		wp_embed_register_handler(
 			self::EMBED_ID,
 			self::EMBED_URL_REGEX,
-			function ( $matches, $attr, $url, $rawattr ) {
-				global $id; // This is the Post ID, [@see WP_Query::setup_postdata, ...]
-				$embed = $this->blade->render(
-					'interactive.shared', [
-						'title' => get_the_title( $id ),
-						'url' => wp_get_shortlink( $id ),
-					]
-				);
-				return apply_filters( 'embed_' . self::EMBED_ID, $embed, $matches, $attr, $url, $rawattr );
-			}
+			[ $this, 'applyEmbedFilterForExport' ]
+		);
+	}
+
+	/**
+	 * Apply embed filter for export
+	 *
+	 * @param $matches
+	 * @param $attr
+	 * @param $url
+	 * @param $rawattr
+	 * @return mixed|void
+	 */
+	public function applyEmbedFilterForExport( $matches, $attr, $url, $rawattr ) {
+		global $id;
+		$this->setIframeCounterByPost( $id );
+		$embed = $this->blade->render(
+			'interactive.media', [
+				'title' => get_the_title( $id ),
+				'url' => wp_get_shortlink( $id ),
+				'tag' => 'iframe',
+				'id' => 'iframe-phet-' . $this->iframes_counter[ $id ],
+			]
 		);
+		apply_filters( 'embed_' . self::EMBED_ID, $embed, $matches, $attr, $url, $rawattr );
+		return $embed;
```

Comment:
Could we do this directly?
```php
return apply_filters( 'embed_' . self::EMBED_ID, $embed, $matches, $attr, $url, $rawattr );
```

Reply:
I added this return to be able to test the method, so far we don't need any return on this method but it will simplify our testing